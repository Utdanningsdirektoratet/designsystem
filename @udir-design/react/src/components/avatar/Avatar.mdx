import { Meta, Canvas, Controls, Primary } from '@storybook/addon-docs/blocks';
import * as AvatarStories from './Avatar.stories';

<Meta of={AvatarStories} />

# Avatar

`Avatar` viser et bilde, initialer eller et ikon for en bruker eller profil.

**Passer til å**

- representere en person eller enhet

**Passer ikke til å**

- representere handlinger, objekter eller steder, bruk heller ikoner eller andre visuelle symboler
- brukes som visuell dekor

## Slik bruker du Avatar

<Primary />
<Controls />

```tsx
import { Avatar } from '@udir-design/react';

/* Med standard brukerikon */
<Avatar aria-label="Ola Nordmann" />

/* Med initialer */
<Avatar aria-label="Ola Nordmann">ON</Avatar>

/* Med bilde */
<Avatar aria-label="Ola Nordmann">
  <img src="..." />
</Avatar>

/* Sammen med synlig tekst */
<>
  <Avatar aria-hidden>ON</Avatar>
  <span>Ola Nordmann</span>
</>
```

## Eksempler

### Innhold

`Avatar` kan inneholde et ikon, et bilde, eller initialer. Ikon og bilde legger du inn som `children` i komponenten. Innholdet får automatisk `aria-hidden="true"` for å unngå dobbel informasjon.
Initialer sendes inn i `initials`-propen. Det vil fungere som en fallback dersom bildet ikke lastes. Et standard ikon vises dersom det ikke sendes inn `children` eller `initials`.

<Canvas of={AvatarStories.Content} />

### Størrelser

`Avatar` kommer i flere størrelser. Tekst og ikon vil skalere, men det kan hende ikonet ditt må tilpasses.

<Canvas of={AvatarStories.Sizes} />

### Farger

Du kan bruke alle temafargene på `Avatar`.

<Canvas of={AvatarStories.ColorVariants} />

### Komponering

Du kan integrere `Avatar` i andre komponenter, for eksempel i en `Dropdown`. I tillegg kan du kombinere den med komponenter som `Badge` for å indikere status.

<Canvas of={AvatarStories.InDropdown} />

### Som lenke

Legg lenker eller knapper rundt `Avatar` for å gjøre den klikkbar.
Du må selv sette styling dersom du skal ha en interaktiv `Avatar`.

<Canvas of={AvatarStories.AsLink} />

## Retningslinjer

`Avatar` skal bidra til å gjøre brukergrensesnittet mer gjenkjennelig og lett å navigere i, uten å forstyrre eller ta for mye plass.

- Bruk samme form på avataren på en konsekvent måte i hele løsningen.
- Avataren bør som regel kombineres med tekst, med mindre det er helt åpenbart hvem eller hva avataren representerer.
- Bruk avatar kun når det faktisk gir verdi for brukeropplevelsen, for eksempel å vise hvem som har skrevet noe, eller hvem som er ansvarlig.

## Tekst i Avatar

`Avatar` skal som hovedregel ikke inneholde tekst i selve komponenten, med unntak av når initialer brukes. Annen tekst, for eksempel fullt navn, rolle eller organisasjon, bør vises utenfor avataren for å sikre god lesbarhet.

## Tilgjengelighet

`Avatar` må enten ha en `aria-label` som forklarer hvem eller hva avataren representerer, eller ha `aria-hidden` og synlig tekst som formidler den samme informasjonen. Ikke spesifiser i `aria-label` at det er et bilde, skjermleseren formidler dette automatisk. Det holder å bruke navnet, som for eksempel `aria-label="Ola Nordmann"`.

Bilder, og andre `children`, får automatisk `aria-hidden="true"` for å unngå dobbel informasjon.
