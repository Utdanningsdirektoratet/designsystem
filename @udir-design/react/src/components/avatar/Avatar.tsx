import {
  Avatar as DigdirAvatar,
  type AvatarProps as DigdirAvatarProps,
} from '@digdir/designsystemet-react';
import type {
  ComponentRef,
  ForwardRefExoticComponent,
  RefAttributes,
} from 'react';

type AvatarProps = Omit<DigdirAvatarProps, 'variant'>;
const Avatar = DigdirAvatar as ForwardRefExoticComponent<
  AvatarProps & RefAttributes<ComponentRef<typeof DigdirAvatar>>
>;

export type { AvatarProps };
export { Avatar };
