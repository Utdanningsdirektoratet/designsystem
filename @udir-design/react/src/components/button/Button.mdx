import { Meta, Canvas, Controls, Primary } from '@storybook/blocks';
import { Do, Dont, Stack } from '../../../.storybook/docs-components';
import { Button } from './Button';
import * as ButtonStories from './Button.stories';
import { ButtonEx1, ButtonEx2, ButtonEx4, ButtonEx5 } from './docs/doAndDont';

<Meta of={ButtonStories} />

# Button

Knapper lar brukerne utføre handlinger. Vi har variantene `primary`, `secondary` og `tertiary` som kommer i ulike fargevarianter og størrelser.

<Primary />
<Controls />

## Slik bruker du button

```tsx
import { Button } from '@udir-design/react';

<Button>Trykk på meg!</Button>;
```

## Varianter av button

Vi har forskjellige knapper for å fremheve utvalgte handlinger. `Primary`, `secondary` og `tertiary` bestemmer hvor viktig handlingen er.

### Primary button

Vi bruker denne knappen til den handlingen som er viktigst for brukeren, for eksempel Lagre, Neste eller Send inn. Det skal bare være en primærknapp per side.

<Canvas of={ButtonStories.Primary} />

### Secondary button

Vi bruker denne knappen til handlinger som ikke er hovedhandlinger. Den står gjerne sammen med en hovedknapp, så hvis hovedknappen er Lagre, kan sekundærknappen være Avbryt.

<Canvas of={ButtonStories.Secondary} />

### Tertiary button

Denne knappen bruker vi til nedtonede handlinger. Hvis vi vil bruke den alene, må den ha med et ikon, hvis ikke er det vanskelig å se at det er en knapp.

<Canvas of={ButtonStories.Tertiary} />

## Fargevarianter

Knappene kommer i to fargevarianter: `neutral` og `danger`.

- `neutral` er standardfargen.

<Canvas of={ButtonStories.Neutral} />

- `danger`-fargen bruker vi til handlinger som brukeren ikke kan angre, for eksempel Slett.

<Canvas of={ButtonStories.Danger} />

### Kombinere ulike fargevarianter

Vi kan bruke en `danger`-knapp sammen med én eller flere `neutral`-knapper når vi skal tydeliggjøre en handling brukeren ikke kan angre, for eksempel Forkast. Vi skal ikke kombinere andre fargevarianter.

<Canvas of={ButtonStories.CombinedColors} />

## Eksempler

### Knapp som lenke

Bruk `asChild` dersom du trenger å bruke knappen som en lenke. Vurder først om det er bedre å bruke [`Link`](/docs/components-link--docs)-komponenten.
Du kan lese mer om komposisjon [her](https://www.designsystemet.no/grunnleggende/for-utviklere/komposisjon).

<Canvas of={ButtonStories.AsLink} />

### Knapp med kun ikon

Knapper med kun ikon bruker vi bare for godt kjente ikoner som Lukk og Slett. Når knappen bare har ikon, må vi ha med en tilgjengelig tekst som forklarer hva knappen gjør. Dette kan legges i `title`, `aria-label` eller i en `Tooltip` som i eksempelet under.

<Canvas of={ButtonStories.IconOnly} />

### Knapp med tekst og ikon

Når vi bruker ikon sammen med tekst, skal ikonet ha `aria-hidden` slik at teksten ikke blir lest opp av skjermleser to ganger.

<Canvas of={ButtonStories.TextAndIcon} />

Vi plasserer ikonet til venstre eller høyre for teksten, avhengig av hva knappen kommuniserer og hvor den står i grensesnittet.

Som regel er det **venstre** som er den beste plasseringen, men det finnes unntak:

- Er det en knapp som sier “Start” med en høyrepekende pil, er det naturlig at pilen plasseres til høyre for teksten.
- Er knappen fast plassert på høyre side av grensesnittet er det best at ikonet også er høyrestilt for teksten.

### Størrelse på ikon

Ikon skalerer automatisk basert på knappestørrelse, men du kan endre denne ved å sette font størrelsen.

<Canvas of={ButtonStories.IconOnly} />

### Last inn

Når vi skal vise brukeren at knappen laster noe, kan vi kombinere knappen med en `spinner`.
I eksempelet under bruker vi proppen `loading` for å vise at knappen laster.

<Canvas of={ButtonStories.Loading} />

I eksempelet over blir `aria-disabled` satt av `loading="true"`. Dette er den anbefalte måten for å vise at en knapp er deaktivert. Da kan den fortsatt få fokus når noen navigerer med tastaturet. Slik vil blant annet skjermlesere og andre hjelpemidler bli informert om at knappen finnes, men ikke er aktiv.

Denne egenskapen stopper ikke automatisk knappen fra å utløse `onClick`.

Pass på at callback-funksjonen ikke kjører når knappen er deaktivert.

## Retningslinjer for button

Knapper skal brukes til å utføre handlinger.

**Passer til å**

- starte en prosess
- navigere i en flyt, for eksempel med Forrige og Neste
- sende inn et skjema

**Passer ikke til å**

- navigere til andre sider eller ut av tjenesten, bruk heller [`link`](/docs/components-link--docs)

### Plassering av knapper

Når vi legger flere knapper ved siden av hverandre, skal primærknappen stå først. Unntaket er Forrige og Neste, da skal sekundærknappen Forrige stå først.

### Ikke bruk deaktiverte knapper

Deaktiverte knapper bør unngås fordi de kan være vanskelige å oppfatte. Noen brukere vil ikke forstå hva knappen sier eller hvorfor den ikke er klikkbar. Bruk heller en aktiv knapp som validerer.

### En knapp skal utføre kun én handling

Vi skal tydelig beskrive handlingene knappene utfører. En knapp kan bare ha én handling, ikke flere.

<ButtonEx1 />

### Èn hovedhandling per side

Vi skal holde oss til én primærknapp per side. Hvis det er flere hovedhandlinger kan brukerne bli usikre på hva de skal gjøre.

<ButtonEx2 />

### Ett ikon per knapp

Vi skal holde oss til ett ikon per knapp.

<ButtonEx4 />

## Tekst i komponenten

Knappeteksten skal være kort, helst ikke mer enn to, maks. tre ord. Tenk på at knappen tydelig skal beskrive hva som skjer når brukeren klikker på den.

### Unngå å blande skrivemåter

Bruk stor forbokstav på det første ordet, som du ville gjort i en setning. Ikke bruk bare store bokstaver (all caps), det er vanskeligere å lese.

<ButtonEx5 />
