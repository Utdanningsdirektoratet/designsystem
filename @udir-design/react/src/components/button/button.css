.ds-button {
  &[data-variant='tertiary']:not([data-color='danger']),
  &[data-variant='secondary']:not([data-color='danger']) {
    --dsc-button-color: var(--ds-color-text-default);
  }
}
