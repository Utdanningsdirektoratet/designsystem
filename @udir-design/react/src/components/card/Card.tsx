import {
  Card,
  CardBlock,
  type CardBlockProps,
  type CardProps,
} from '@digdir/designsystemet-react';

// For some reason this fixes "ComponentSubcomponent" -> "Component.Subcomponent" in Storybook code snippets
Card.displayName = 'Card';

export { Card, CardBlock, CardBlockProps, CardProps };
