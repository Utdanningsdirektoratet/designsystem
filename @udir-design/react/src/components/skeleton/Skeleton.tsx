import {
  Skeleton as DigdirSkeleton,
  type SkeletonProps as DigdirSkeletonProps,
} from '@digdir/designsystemet-react';
import { ComponentRef, RefAttributes } from 'react';

type SkeletonProps = Omit<DigdirSkeletonProps, 'characters'>;

const Skeleton: React.ForwardRefExoticComponent<
  SkeletonProps & RefAttributes<ComponentRef<typeof DigdirSkeleton>>
> = DigdirSkeleton;

export { Skeleton };
export type { SkeletonProps };
