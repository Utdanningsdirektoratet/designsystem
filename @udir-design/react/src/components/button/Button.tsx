import {
  Button as DigdirButton,
  type ButtonProps as DigdirButtonProps,
} from '@digdir/designsystemet-react';
import { ComponentRef, ForwardRefExoticComponent, RefAttributes } from 'react';
import './button.css';

type ButtonProps = Omit<DigdirButtonProps, 'data-color'> & {
  /**
   * Change the color scheme of the button
   */
  'data-color'?: 'neutral' | 'danger';
};

const Button = DigdirButton as ForwardRefExoticComponent<
  ButtonProps & RefAttributes<ComponentRef<typeof DigdirButton>>
>;

export { Button };
export type { ButtonProps };
