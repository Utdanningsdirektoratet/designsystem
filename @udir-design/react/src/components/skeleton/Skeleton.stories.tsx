import type { Meta, StoryObj } from '@storybook/react-vite';
import {
  ArrowsUpDownIcon,
  FilterIcon,
  PlusCircleIcon,
} from '@udir-design/icons';
import { Skeleton } from './Skeleton';
import { Paragraph } from '../typography/paragraph/Paragraph';
import { Heading } from '../typography/heading/Heading';
import { Button } from '../button/Button';
import { Card } from '../card/Card';
import { Avatar } from '../avatar/Avatar';

const meta: Meta<typeof Skeleton> = {
  component: Skeleton,
  tags: ['beta', 'digdir'],
  parameters: {
    componentOrigin: {
      originator: 'digdir',
    },
    layout: 'centered',
    a11y: {
      config: {
        // Disable a11y empty heading rule as we intentionally set aria-hidden="true" on the Skeleton component inside Headings
        rules: [{ id: 'empty-heading', selector: ':has(.ds-skeleton)' }],
      },
    },
  },
};

export default meta;
type Story = StoryObj<typeof Skeleton>;

export const Preview: Story = {
  args: {
    width: 200,
    height: 100,
  },
};

export const Components: Story = {
  args: {
    height: 50,
  },
  render: (args) => {
    return (
      <div
        style={{
          display: 'flex',
          alignItems: 'center',
          gap: '20px',
        }}
      >
        <Skeleton variant="circle" width="50px" {...args} />
        <Skeleton variant="rectangle" width="100px" {...args} />
        <Paragraph>
          <Skeleton variant="text" width="10" {...args} />
        </Paragraph>
      </div>
    );
  },
};

const StudentData = [
  {
    id: 0,
    name: 'Rita Nordmann',
    birthday: '12.09.2010',
    class: '5.-7. trinn',
    teacher: 'Nanna Arnestad',
  },
  {
    id: 1,
    name: 'Ola Nordmann',
    birthday: '11.03.2010',
    class: '5.-7. trinn',
    teacher: 'Nanna Arnestad',
  },
  {
    id: 2,
    name: 'Kari Nordmann',
    birthday: '04.12.2010',
    class: '5.-7. trinn',
    teacher: 'Nanna Arnestad',
  },
  {
    id: 3,
    name: 'Juliane Nordmann',
    birthday: '24.04.2010',
    class: '5.-7. trinn',
    teacher: 'Nanna Arnestad',
  },
  {
    id: 4,
    name: 'Mateo Nordmann',
    birthday: '01.05.2010',
    class: '5.-7. trinn',
    teacher: 'Nanna Arnestad',
  },
  {
    id: 5,
    name: 'Kai Nordmann',
    birthday: '17.05.2010',
    class: '5.-7. trinn',
    teacher: 'Nanna Arnestad',
  },
];

export const Cards: Story = {
  render: (args) => {
    return (
      <div
        style={{
          margin: 'var(--ds-size-12)',
          display: 'flex',
          flexDirection: 'column',
          gap: 'var(--ds-size-6)',
        }}
      >
        <Heading level={3} data-size="lg">
          Elever
        </Heading>
        <div style={{ display: 'flex', justifyContent: 'end' }}>
          <Button variant="tertiary">
            <PlusCircleIcon aria-hidden /> Legg til elev
          </Button>
          <Button variant="tertiary">
            <ArrowsUpDownIcon aria-hidden /> Sorter
          </Button>
          <Button variant="tertiary">
            <FilterIcon aria-hidden /> Filter
          </Button>
        </div>
        <div
          style={{
            display: 'grid',
            gridTemplateColumns: 'repeat(3, 1fr)',
            gap: 'var(--ds-size-6)',
          }}
        >
          {StudentData.map((student) => (
            <Card key={student.id}>
              <Card.Block>
                <Skeleton variant="text">{student.name}</Skeleton>
              </Card.Block>
              <Card.Block style={{ display: 'flex', flexDirection: 'column' }}>
                <Skeleton variant="text">
                  Fødselsdato: {student.birthday}
                </Skeleton>
                <Skeleton variant="text">Trinn: {student.class}</Skeleton>
                <Skeleton variant="text">Lærer: {student.teacher}</Skeleton>
                <Skeleton variant="text">Vurdert: 23.04.2023</Skeleton>
              </Card.Block>
            </Card>
          ))}
        </div>
      </div>
    );
  },
};

const schoolImage =
  'https://images.unsplash.com/photo-1516383607781-913a19294fd1?q=80&w=2374&auto=format&fit=crop&ixlib=rb-4.1.0&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D';

export const UsageExample: Story = {
  render: (args) => {
    return (
      <div
        style={{
          display: 'flex',
          gap: 'var(--ds-size-12)',
          margin: 'var(--ds-size-12)',
        }}
      >
        <div style={{ flex: '1 1 200px' }}>
          <Skeleton height="200px" {...args} />
          <Heading>
            <Skeleton variant="text" width="15" />
          </Heading>
          <div
            style={{
              display: 'flex',
              gap: '10px',
              alignItems: 'center',
              padding: '5px 0 5px 0',
            }}
          >
            <Skeleton variant="circle" width="40px" height="40px" {...args} />
            <Skeleton variant="text" {...args}>
              Forfatter| sist endret
            </Skeleton>
          </div>
          <Skeleton variant="text" width="200" />
        </div>
        <div style={{ flex: '1 1 200px' }}>
          <img
            src={schoolImage}
            alt="Bøker på et bord"
            style={{
              width: '100%',
              height: '200px',
              objectFit: 'cover',
              borderRadius: 10,
            }}
          />
          <Heading>Undersøke</Heading>
          <div style={{ display: 'flex', alignItems: 'center', gap: '10px' }}>
            <Avatar data-size="xs" initials="KN" aria-label={'Kari Nordmann'} />
            <Paragraph>Kari Nordmann | 27.06.2025</Paragraph>
          </div>
          <Paragraph>
            Barnehagen er pliktig til å gjennomføre undersøkelser som med
            rimelighet kan forventes. Barnehagen må innhente nok informasjon til
            å avdekke hva som er problemet og hva som gjør at barnet ikke har
            det trygt og godt.
          </Paragraph>
        </div>
      </div>
    );
  },
};

export const Children: Story = {
  args: {
    variant: 'rectangle',
    children: [
      <Paragraph>
        Her er en tekst som blir sendt inn som barn av en Skeleton.
      </Paragraph>,
      <Paragraph>
        Se hvordan Skeleton da dekker den samlede bredden og høyden til barna.
      </Paragraph>,
      <Button>Knapp</Button>,
    ],
  },
};

export const Text: Story = {
  args: {
    variant: 'text',
  },
  render: (args) => (
    <div
      style={{
        display: 'flex',
        gap: 'var(--ds-size-12)',
        margin: 'var(--ds-size-12)',
      }}
    >
      <div style={{ flex: '1 1 200px' }}>
        <Heading>Den kjekkaste leiken</Heading>
        <Paragraph data-size="sm">
          Det er verken dyrt eller komplisert å leggje til rette for den leiken
          barna liker best. Når viser barna mest trivsel i uteleiken? Det ønskte
          forskarar i forskingsprosjektet «EnCompetence» å finne ut, og dei
          ville spesielt sjå på situasjonar med fysisk aktivitet.
        </Paragraph>
      </div>
      <div style={{ flex: '1 1 200px' }}>
        <Heading>
          <Skeleton {...args}>Den kjekkaste leiken</Skeleton>
        </Heading>
        <Paragraph data-size="sm">
          <Skeleton width={262} {...args} />
        </Paragraph>
      </div>
    </div>
  ),
};
