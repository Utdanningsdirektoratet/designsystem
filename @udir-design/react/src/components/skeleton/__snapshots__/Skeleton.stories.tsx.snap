// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Cards 1`] = `
<div style="margin: var(--ds-size-12); display: flex; flex-direction: column; gap: var(--ds-size-6);">
  <h3 data-size="lg">
    Elever
  </h3>
  <div style="display: flex; justify-content: end;">
    <button
      data-variant="tertiary"
      type="button"
    >
      <svg
        xmlns="http://www.w3.org/2000/svg"
        width="1em"
        height="1em"
        fill="none"
        viewbox="0 0 24 24"
        focusable="false"
        role="img"
        aria-hidden="true"
      >
        <path
          fill="currentColor"
          fill-rule="evenodd"
          d="M3.75 12a8.25 8.25 0 1 1 16.5 0 8.25 8.25 0 0 1-16.5 0M12 2.25c-5.385 0-9.75 4.365-9.75 9.75s4.365 9.75 9.75 9.75 9.75-4.365 9.75-9.75S17.385 2.25 12 2.25m0 4.5a.75.75 0 0 1 .75.75v3.75h3.75a.75.75 0 0 1 0 1.5h-3.75v3.75a.75.75 0 0 1-1.5 0v-3.75H7.5a.75.75 0 0 1 0-1.5h3.75V7.5a.75.75 0 0 1 .75-.75"
          clip-rule="evenodd"
        >
        </path>
      </svg>
      Legg til elev
    </button>
    <button
      data-variant="tertiary"
      type="button"
    >
      <svg
        xmlns="http://www.w3.org/2000/svg"
        width="1em"
        height="1em"
        fill="none"
        viewbox="0 0 24 24"
        focusable="false"
        role="img"
        aria-hidden="true"
      >
        <path
          fill="currentColor"
          fill-rule="evenodd"
          d="M16.47 21.03a.75.75 0 0 0 1.06 0l3.5-3.5a.75.75 0 1 0-1.06-1.06l-2.22 2.22V9.5a.75.75 0 0 0-1.5 0v9.19l-2.22-2.22a.75.75 0 1 0-1.06 1.06zM4.03 7.53l2.22-2.22v9.19a.75.75 0 0 0 1.5 0V5.31l2.22 2.22a.75.75 0 1 0 1.06-1.06l-3.5-3.5a.75.75 0 0 0-1.06 0l-3.5 3.5a.75.75 0 0 0 1.06 1.06"
          clip-rule="evenodd"
        >
        </path>
      </svg>
      Sorter
    </button>
    <button
      data-variant="tertiary"
      type="button"
    >
      <svg
        xmlns="http://www.w3.org/2000/svg"
        width="1em"
        height="1em"
        fill="none"
        viewbox="0 0 24 24"
        focusable="false"
        role="img"
        aria-hidden="true"
      >
        <path
          fill="currentColor"
          fill-rule="evenodd"
          d="M3.25 7A.75.75 0 0 1 4 6.25h16a.75.75 0 0 1 0 1.5H4A.75.75 0 0 1 3.25 7m3 5a.75.75 0 0 1 .75-.75h10a.75.75 0 0 1 0 1.5H7a.75.75 0 0 1-.75-.75M10 16.25a.75.75 0 0 0 0 1.5h4a.75.75 0 0 0 0-1.5z"
          clip-rule="evenodd"
        >
        </path>
      </svg>
      Filter
    </button>
  </div>
  <div style="display: grid; grid-template-columns: repeat(3, 1fr); gap: var(--ds-size-6);">
    <div data-variant="default">
      <div>
        <span
          aria-hidden="true"
          data-text="-"
          data-variant="text"
        >
          Rita Nordmann
        </span>
      </div>
      <div style="display: flex; flex-direction: column;">
        <span
          aria-hidden="true"
          data-text="-"
          data-variant="text"
        >
          Fødselsdato: 12.09.2010
        </span>
        <span
          aria-hidden="true"
          data-text="-"
          data-variant="text"
        >
          Trinn: 5.-7. trinn
        </span>
        <span
          aria-hidden="true"
          data-text="-"
          data-variant="text"
        >
          Lærer: Nanna Arnestad
        </span>
        <span
          aria-hidden="true"
          data-text="-"
          data-variant="text"
        >
          Vurdert: 23.04.2023
        </span>
      </div>
    </div>
    <div data-variant="default">
      <div>
        <span
          aria-hidden="true"
          data-text="-"
          data-variant="text"
        >
          Ola Nordmann
        </span>
      </div>
      <div style="display: flex; flex-direction: column;">
        <span
          aria-hidden="true"
          data-text="-"
          data-variant="text"
        >
          Fødselsdato: 11.03.2010
        </span>
        <span
          aria-hidden="true"
          data-text="-"
          data-variant="text"
        >
          Trinn: 5.-7. trinn
        </span>
        <span
          aria-hidden="true"
          data-text="-"
          data-variant="text"
        >
          Lærer: Nanna Arnestad
        </span>
        <span
          aria-hidden="true"
          data-text="-"
          data-variant="text"
        >
          Vurdert: 23.04.2023
        </span>
      </div>
    </div>
    <div data-variant="default">
      <div>
        <span
          aria-hidden="true"
          data-text="-"
          data-variant="text"
        >
          Kari Nordmann
        </span>
      </div>
      <div style="display: flex; flex-direction: column;">
        <span
          aria-hidden="true"
          data-text="-"
          data-variant="text"
        >
          Fødselsdato: 04.12.2010
        </span>
        <span
          aria-hidden="true"
          data-text="-"
          data-variant="text"
        >
          Trinn: 5.-7. trinn
        </span>
        <span
          aria-hidden="true"
          data-text="-"
          data-variant="text"
        >
          Lærer: Nanna Arnestad
        </span>
        <span
          aria-hidden="true"
          data-text="-"
          data-variant="text"
        >
          Vurdert: 23.04.2023
        </span>
      </div>
    </div>
    <div data-variant="default">
      <div>
        <span
          aria-hidden="true"
          data-text="-"
          data-variant="text"
        >
          Juliane Nordmann
        </span>
      </div>
      <div style="display: flex; flex-direction: column;">
        <span
          aria-hidden="true"
          data-text="-"
          data-variant="text"
        >
          Fødselsdato: 24.04.2010
        </span>
        <span
          aria-hidden="true"
          data-text="-"
          data-variant="text"
        >
          Trinn: 5.-7. trinn
        </span>
        <span
          aria-hidden="true"
          data-text="-"
          data-variant="text"
        >
          Lærer: Nanna Arnestad
        </span>
        <span
          aria-hidden="true"
          data-text="-"
          data-variant="text"
        >
          Vurdert: 23.04.2023
        </span>
      </div>
    </div>
    <div data-variant="default">
      <div>
        <span
          aria-hidden="true"
          data-text="-"
          data-variant="text"
        >
          Mateo Nordmann
        </span>
      </div>
      <div style="display: flex; flex-direction: column;">
        <span
          aria-hidden="true"
          data-text="-"
          data-variant="text"
        >
          Fødselsdato: 01.05.2010
        </span>
        <span
          aria-hidden="true"
          data-text="-"
          data-variant="text"
        >
          Trinn: 5.-7. trinn
        </span>
        <span
          aria-hidden="true"
          data-text="-"
          data-variant="text"
        >
          Lærer: Nanna Arnestad
        </span>
        <span
          aria-hidden="true"
          data-text="-"
          data-variant="text"
        >
          Vurdert: 23.04.2023
        </span>
      </div>
    </div>
    <div data-variant="default">
      <div>
        <span
          aria-hidden="true"
          data-text="-"
          data-variant="text"
        >
          Kai Nordmann
        </span>
      </div>
      <div style="display: flex; flex-direction: column;">
        <span
          aria-hidden="true"
          data-text="-"
          data-variant="text"
        >
          Fødselsdato: 17.05.2010
        </span>
        <span
          aria-hidden="true"
          data-text="-"
          data-variant="text"
        >
          Trinn: 5.-7. trinn
        </span>
        <span
          aria-hidden="true"
          data-text="-"
          data-variant="text"
        >
          Lærer: Nanna Arnestad
        </span>
        <span
          aria-hidden="true"
          data-text="-"
          data-variant="text"
        >
          Vurdert: 23.04.2023
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`Children 1`] = `
<span
  aria-hidden="true"
  data-variant="rectangle"
>
  <p data-variant="default">
    Her er en tekst som blir sendt inn som barn av en Skeleton.
  </p>
  <p data-variant="default">
    Se hvordan Skeleton da dekker den samlede bredden og høyden til barna.
  </p>
  <button
    data-variant="primary"
    type="button"
  >
    Knapp
  </button>
</span>
`;

exports[`Components 1`] = `
<div style="display: flex; align-items: center; gap: 20px;">
  <span
    aria-hidden="true"
    data-variant="circle"
    style="width: 50px; height: 50px;"
  >
  </span>
  <span
    aria-hidden="true"
    data-variant="rectangle"
    style="width: 100px; height: 50px;"
  >
  </span>
  <p data-variant="default">
    <span
      aria-hidden="true"
      data-text="----------"
      data-variant="text"
    >
    </span>
  </p>
</div>
`;

exports[`Preview 1`] = `
<span
  aria-hidden="true"
  data-variant="rectangle"
  style="width: 200px; height: 100px;"
>
</span>
`;

exports[`Text 1`] = `
<div style="display: flex; gap: var(--ds-size-12); margin: var(--ds-size-12);">
  <div style="flex: 1 1 200px;">
    <h2>
      Den kjekkaste leiken
    </h2>
    <p
      data-variant="default"
      data-size="sm"
    >
      Det er verken dyrt eller komplisert å leggje til rette for den leiken barna liker best. Når viser barna mest trivsel i uteleiken? Det ønskte forskarar i forskingsprosjektet «EnCompetence» å finne ut, og dei ville spesielt sjå på situasjonar med fysisk aktivitet.
    </p>
  </div>
  <div style="flex: 1 1 200px;">
    <h2>
      <span
        aria-hidden="true"
        data-text="-"
        data-variant="text"
      >
        Den kjekkaste leiken
      </span>
    </h2>
    <p
      data-variant="default"
      data-size="sm"
    >
      <span
        aria-hidden="true"
        data-text="----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------"
        data-variant="text"
      >
      </span>
    </p>
  </div>
</div>
`;

exports[`Usage Example 1`] = `
<div style="display: flex; gap: var(--ds-size-12); margin: var(--ds-size-12);">
  <div style="flex: 1 1 200px;">
    <span
      aria-hidden="true"
      data-variant="rectangle"
      style="height: 200px;"
    >
    </span>
    <h2>
      <span
        aria-hidden="true"
        data-text="---------------"
        data-variant="text"
      >
      </span>
    </h2>
    <div style="display: flex; gap: 10px; align-items: center; padding: 5px 0px;">
      <span
        aria-hidden="true"
        data-variant="circle"
        style="width: 40px; height: 40px;"
      >
      </span>
      <span
        aria-hidden="true"
        data-text="-"
        data-variant="text"
      >
        Forfatter| sist endret
      </span>
    </div>
    <span
      aria-hidden="true"
      data-text="--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------"
      data-variant="text"
    >
    </span>
  </div>
  <div style="flex: 1 1 200px;">
    <img
      alt="Bøker på et bord"
      src="https://images.unsplash.com/photo-1516383607781-913a19294fd1?q=80&amp;w=2374&amp;auto=format&amp;fit=crop&amp;ixlib=rb-4.1.0&amp;ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D"
      style="width: 100%; height: 200px; object-fit: cover; border-radius: 10px;"
    >
    <h2>
      Undersøke
    </h2>
    <div style="display: flex; align-items: center; gap: 10px;">
      <span
        data-variant="circle"
        data-initials="KN"
        role="img"
        aria-label="Kari Nordmann"
        data-size="xs"
      >
      </span>
      <p data-variant="default">
        Kari Nordmann | 27.06.2025
      </p>
    </div>
    <p data-variant="default">
      Barnehagen er pliktig til å gjennomføre undersøkelser som med rimelighet kan forventes. Barnehagen må innhente nok informasjon til å avdekke hva som er problemet og hva som gjør at barnet ikke har det trygt og godt.
    </p>
  </div>
</div>
`;
